templates/CRM/Member/Form/Membership.tpl

```diff
@@ -174,10 +174,10 @@
         {/if}
         {include file="CRM/Member/Form/MembershipCommon.tpl"}
 
-        {if $emailExists and $isEmailEnabledForSite}
+        {if $billingEmailExists and $isEmailEnabledForSite}
           <tr id="send-receipt" class="crm-membership-form-block-send_receipt">
             <td class="label">{$form.send_receipt.label}</td><td>{$form.send_receipt.html}<br />
-            <span class="description">{ts 1=$emailExists}Automatically email a membership confirmation and receipt to %1? OR if the payment is from a different contact, this email will only go to them.{/ts}</span></td>
+            <span class="description">{ts 1=$billingEmailExists}Automatically email a membership confirmation and receipt to %1? OR if the payment is from a different contact, this email will only go to them.{/ts}</span></td>
```

Comment:
@agileware Pushed the changes, Billing email address will be used to show while adding new contribution, if not found then it would fall back to primary email address.